src/test/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java

```diff
@@ -46,14 +61,18 @@ void should_remove_only_files_that_are_old_enough() throws Exception {
         // when
         new DeleteOldFilesTask(ftp, serviceFolderMapping, ttl).run();
 
+        // and
+        verify(ftp).runWith(captureRunWith.capture());
+        captureRunWith.getValue().apply(sftpClient);
```

Comment:
Why are we executing this function here?

Reply:
Because `ftp` is a mock and we actually want to run delete function which is now wrapped under `runWith` and not the other way

Reply:
having setup as it is now and providing invocation answers would lead to _unchecked cats_ warning and code diff from now as follows:

```diff
diff --git a/src/test/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java b/src/test/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java
index 91f22a8..913fe08 100644
--- a/src/test/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java
+++ b/src/test/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java
@@ -6,8 +6,6 @@ import net.schmizz.sshj.sftp.SFTPClient;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Captor;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.hmcts.reform.sendletter.exception.FtpException;
@@ -36,11 +34,11 @@ class DeleteOldFilesTaskTest {
     @Mock private SFTPClient sftpClient;
     @Mock private ServiceFolderMapping serviceFolderMapping;
 
-    @Captor private ArgumentCaptor<Function<SFTPClient, Void>> captureRunWith;
-
     @BeforeEach
     void setUp() {
-        given(ftp.runWith(any())).willReturn(0);
+        given(ftp.runWith(any())).willAnswer(invocation ->
+            ((Function<SFTPClient, Void>) invocation.getArgument(0)).apply(sftpClient)
+        );
     }
 
     @Test
@@ -61,11 +59,8 @@ class DeleteOldFilesTaskTest {
         // when
         new DeleteOldFilesTask(ftp, serviceFolderMapping, ttl).run();
 
-        // and
-        verify(ftp).runWith(captureRunWith.capture());
-        captureRunWith.getValue().apply(sftpClient);
-
         // then
+        verify(ftp).runWith(any());
         verify(ftp).deleteFile("old.zip", sftpClient);
         verify(ftp, never()).deleteFile("new.zip", sftpClient);
         verify(ftp, never()).deleteFile("almostOld.zip", sftpClient);
@@ -86,11 +81,8 @@ class DeleteOldFilesTaskTest {
         // when
         new DeleteOldFilesTask(ftp, serviceFolderMapping, Duration.ZERO).run();
 
-        // and
-        verify(ftp, times(2)).runWith(captureRunWith.capture());
-        captureRunWith.getAllValues().forEach(function -> function.apply(sftpClient));
-
         // then
+        verify(ftp, times(2)).runWith(any());
         verify(ftp).deleteFile("a.zip", sftpClient);
         verify(ftp).deleteFile("b.zip", sftpClient);
     }
@@ -116,11 +108,8 @@ class DeleteOldFilesTaskTest {
         // when
         new DeleteOldFilesTask(ftp, serviceFolderMapping, Duration.ZERO).run();
 
-        // and
-        verify(ftp).runWith(captureRunWith.capture());
-        captureRunWith.getValue().apply(sftpClient);
-
         // then
+        verify(ftp).runWith(any());
         files.forEach(file -> verify(ftp).deleteFile(file.path, sftpClient));
     }
```

i don't see a problem right now but if we want a warning to get in - be my guests and suggest the diff looks good to apply ;)

Reply:
I'd vote for this change, but let's hear others.

Reply:
looks good 👍

Reply:
done 👨‍🏫 👩‍🏫